pkg/pillar/pubsub/pubsub.go

```diff
@@ -1171,13 +1175,24 @@ func handleModify(ctxArg interface{}, key string, item interface{}) {
 	} else {
 		log.Debugf("pubsub.handleModify(%s) add %+v for key %s\n",
 			name, newItem, key)
+		create = true
 	}
 	sub.km.key.Store(key, newItem)
 	if log.GetLevel() == log.DebugLevel {
 		sub.dump("after handleModify")
 	}
-	if sub.ModifyHandler != nil {
-		(sub.ModifyHandler)(sub.userCtx, key, newItem)
+	if create == true {
```

Comment:
Why so much code? Makes it harder to read without adding anything.
It is sufficient to replace the whole block with
         if created and sub.CreateHandler != nil {
			(sub.CreateHandler)(sub.userCtx, key, newItem)
	} else if sub.ModifyHandler != nil {
			(sub.ModifyHandler)(sub.userCtx, key, newItem)
		}

Reply:
I thought the same! But @kalyan-nidumolu was able to point out to me that existing applications which do not use the Create Handler would crash because the create hander would be nil. So just in case they want to create and have a nil create handler I added that code.